packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -887,69 +900,167 @@ export class MergeTree {
 	 * @param segments - An array of (not necessarily contiguous) segments with increasing ordinals.
 	 */
 	private slideAckedRemovedSegmentReferences(segments: ISegment[]) {
-		// References are slid in groups to preserve their order.
-		let currentSlideDestination: ISegment | undefined;
-		let currentSlideIsForward: boolean | undefined;
+		function anyLocalRef(
```

Comment:
this and `filterLocalRefs` can be moved to the upper scope. i'd go ahead and do so, either in this file or nearby localReferenceCollection (and import them here). moving as much complexity away from this as we can is ideal.